Improve 'Types and Specs' introduction (#1643)
elixir-lang/elixir-lang.github.com: getting-started/typespecs-and-behaviours.markdown

## Types and specs

> Elixir is a dynamically typed language. Types are not checked, and type errors are not thrown, at compile-time. Nonetheless, Elixir comes with **typespecs**, which are a notation used for:

Reviewer: ```suggestion
Elixir is a dynamically typed language, so all types in Elixir are checked at runtime. Nonetheless, Elixir comes with **typespecs**, which are a notation used for:
```

1. declaring typed function signatures (also called specifications);
2. declaring custom types.

> Although typespecs are primarily for documentation (i.e. they are not enforced, either at compile-time or run-time), they are useful for code clarity and static code analysis (e.g. Erlang's [Dialyzer](http://www.erlang.org/doc/man/dialyzer.html) tool).

Reviewer: ```suggestion
Typespecs are useful for code clarity and static code analysis (for example, Erlang's [Dialyzer](http://www.erlang.org/doc/man/dialyzer.html) tool).
```
